### PR TITLE
Add badges on profile page

### LINKED ISSUE #3395: Display badges in the Profile page
Following the backend work from #3394, the respective badges for a user should be displayed on the Profile page.

The designs of the badges have been sent over in a Figma file by an internal designer. The badges should be displayed in the manner shown [in this section](https://github.com/mozilla/pontoon/blob/3b840b967932e898b4ed398f7c7ea3d847106dd2/specs/0119-gamification-badges.md#mockups) of the spec.

### PR DESCRIPTION
**Pull request by harmitgoswami**
Fix #3395 

This PR displays badges in the user profile page. The respective badges are displayed if the user's level for that badge is above 1, and the level of the badge is displayed on the bottom left of the badge.

The initial commit for this PR is marked as a draft, to hopefully get direction on how to close the gap between the badge and border on the right side and bottom (see image below). 

This PR was branched off from the feature branch in #3413, and will continue the work being done in that PR.

**To reproduce results**: Upon starting the server, head to `localhost:8000/contributors/{user}`, where user is the username of anyone who has enough actions after October 18th (`BADGES_START_DATE` in `base/models/user.py`) to award them a badge based on the [thresholds defined in the spec](https://github.com/mozilla/pontoon/blob/3b840b967932e898b4ed398f7c7ea3d847106dd2/specs/0119-gamification-badges.md#feature-explanation). The badges for the user will be displayed on the left-hand column, below their contact information and above their dates data.

image

### CODE REVIEW

**Review by mathjazz — COMMENTED** [review]
Nice work! Left some suggestions for improvements.

File: pontoon/settings/base.py [hunk 1]
```diff
@@ -961,6 +963,45 @@ def _default_from_email():
 # system time zone.
 TIME_ZONE = os.environ.get("TZ", "UTC")
 
+# Information regarding badge thresholds and data collection
+badges_start_date = os.environ.get("BADGES_START_DATE", "2024-10-18")
```
[the remark this answers is not in the document]
Comment (harmitgoswami) [reply pontoon/settings/base.py:967]: Also while we're at it, what's the best place to put `BADGES_START_DATE`, `TRANSLATION_BADGE_THRESHOLDS`, and `COMMUNITY_BADGE_THRESHOLDS` within `base.py`? I just noticed they're currently in the Internationalization section but that definitely doesn't make sense haha

**Comment by harmitgoswami** [comment]
I went with using Epoch time as the default for `BADGES_START_DATE`, but let me know if you still think we should use `None` instead.

I've also renamed the badges to be a little more specific.

File: pontoon/contributors/templates/contributors/profile.html [hunk 2]
```diff
@@ -121,6 +122,27 @@ <h2 class="display-name">{{ contributor.first_name }}</h2>
...
 
+            {% if has_badges %}
+            <div class="achievements">
+                <div class="block">
+                    <h4 class="subtitle">Achievements</h4>
+                </div>
+                <div class="block">
+                    {% for path, badge in badges.items() %}
+                    <div class="badge-wrapper">
+                        {% if badge.level %}
```
Comment (mathjazz) [inline pontoon/contributors/templates/contributors/profile.html:133]: Move this if block around the `.badge-wrapper`.

File: pontoon/settings/base.py [hunk 3]
```diff
@@ -1111,4 +1113,24 @@ def account_username(user):
...
+    BADGES_START_DATE = timezone.make_aware(
+        datetime.strptime(badges_start_date, "%Y-%m-%d"), timezone=timezone.utc
+    )
+except ValueError as e:
+    raise ValueError(f"Error: {e}")
+
+# Used for Review Master and Translation Champion
+BADGES_TRANSLATION_THRESHOLDS = list(
+    map(
+        int, os.environ.get("BADGES_TRANSLATION_THRESHOLDS", "5,50,250,1000").split(",")
```
Comment (mathjazz) [inline pontoon/settings/base.py:1128]: For consistency:
```suggestion
        int, os.environ.get("BADGES_TRANSLATION_THRESHOLDS", "5, 50, 250, 1000").split(",")
```

**Review by mathjazz — COMMENTED** [review]
Almost there!

Please apply the following diff:

```diff
diff --git a/pontoon/contributors/static/css/profile.css b/pontoon/contributors/static/css/profile.css
index 926eb0a76..a09cf8f9d 100644
--- a/pontoon/contributors/static/css/profile.css
+++ b/pontoon/contributors/static/css/profile.css
@@ -119,27 +119,31 @@ h2 {
   width: 100%;
 }

+.achievements .block {
+  font-size: 0;
+}
+
 .achievements .badge-wrapper {
   position: relative;
   display: inline-block;
+  margin: 10px 7px 0 0;
+}
+
+.achievements .badge-wrapper:last-child {
+  margin-right: 0;
 }

 .achievements img.badge {
-  max-width: 70px;
-  border: 3px solid var(--main-border-1);
-  border-radius: 6px;
-  margin-left: 0.125rem;
+  border: 2px solid var(--main-border-1);
+  border-radius: 50%;
 }

 .achievements .badge-level {
   position: absolute;
-  bottom: 0;
-  left: 0;
-  transform: translate(10%, -30%);
-  background: var(--background-1);
-  color: var(--light-grey-6);
+  bottom: 5px;
+  left: 2px;
+  background: var(--icon-background-1);
   padding: 2px 4px;
-  margin-left: 0.125rem;
   border-radius: 4px;
   font-size: 12px;
 }
diff --git a/pontoon/contributors/templates/contributors/profile.html b/pontoon/contributors/templates/contributors/profile.html
index fb8830c63..57c56caab 100644
--- a/pontoon/contributors/templates/contributors/profile.html
+++ b/pontoon/contributors/templates/contributors/profile.html
@@ -126,8 +126,6 @@
             <div class="achievements">
                 <div class="block">
                     <h4 class="subtitle">Achievements</h4>
-                </div>
-                <div class="block">
                     {% for path, badge in badges.items() %}
                     <div class="badge-wrapper">
                         {% if badge.level %}
```

File: pontoon/settings/base.py [hunk 4]
```diff
@@ -1111,4 +1113,25 @@ def account_username(user):
 # management command will run.
 SUGGESTION_NOTIFICATIONS_DAY = os.environ.get("SUGGESTION_NOTIFICATIONS_DAY", 4)
 
+# Date from which badge data collection starts
+badges_start_date = os.environ.get("BADGES_START_DATE", "2024-10-01")
```
Comment (mathjazz) [inline pontoon/settings/base.py:1117]: The default date is wrong again. :)

**Review by mathjazz — APPROVED** [review]
Great work!

**Pull request merged**